src/Styleguide/Components/CountdownTimer.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  color,
+  Flex,
+  Sans,
+  space,
+  Spacer,
+  StackableBorderBox,
+} from "@artsy/palette"
+import { WithCurrentTime } from "Components/WithCurrentTime"
+import moment from "moment-timezone"
+import React from "react"
+import styled from "styled-components"
+import { TimerIcon } from "Styleguide/Elements/icons/TimerIcon"
+
+const FIVE_HOURS_IN_SECONDS = 60 * 60 * 5
+
+const pad = (n: number) => n.toString().padStart(2, "0")
+
+const TimeRemaining: React.SFC<{
+  currentTime: string
+  countdownEnd: string
+  highlight: Parameters<typeof color>[0]
+}> = ({ currentTime, countdownEnd, highlight = "purple100" }) => {
+  const timeRemaining = moment.duration(
+    moment(countdownEnd).diff(moment(currentTime))
+  )
+  return (
+    <Sans size="3" color={highlight} weight="medium">
+      {Math.floor(timeRemaining.asSeconds()) <= 0 ? (
+        "0 days"
+      ) : (
+        <>
+          {timeRemaining.days() > 0 && pad(timeRemaining.days()) + "d "}
+          {timeRemaining.hours() > 0 && pad(timeRemaining.hours()) + "h "}
+          {timeRemaining.minutes() > 0 && pad(timeRemaining.minutes()) + "m "}
+          {pad(timeRemaining.seconds()) + "s"}
+        </>
+      )}
+      <span> left</span>
+    </Sans>
+  )
+}
+
+const ProgressBarBackground = styled.div`
+  height: 3px;
+  background-color: ${color("black10")};
+  margin: ${space(0.5)}px 0 ${space(1)}px;
+  display: flex;
+  justify-content: flex-start;
+`
+
+const ProgressBar: React.SFC<{
+  currentTime: string
+  countdownStart: string
+  countdownEnd: string
+  highlight: Parameters<typeof color>[0]
+}> = ({
+  currentTime,
+  countdownStart,
+  countdownEnd,
+  highlight = "purple100",
+}) => {
+  const secondsRemaining = moment(countdownEnd).diff(
+    moment(currentTime),
+    "seconds"
+  )
+  const secondsSinceStart = moment(currentTime).diff(
```

Comment:
👍 in my head, I assumed you'd need `secondsRemaining`, and `totalSeconds` for the width calc. I see you derive `totalSeconds` a tad indirectly, by doing `secondsSinceStart + secondsRemaining`. That works!

Reply:
oh yeah that's easier to grok, I'll switch to that. I think at one point I was using `secondsSinceStart` in some other context too.